config/initializers/devise.rb

```diff
@@ -257,3 +257,7 @@
   # so you need to do it manually. For the users scope, it would be:
   # config.omniauth_path_prefix = '/my_engine/users/auth'
 end
+
+Rails.application.config.to_prepare do
+  Devise::RegistrationsController.layout proc { |controller| user_signed_in? ? "dashboard" : "application" }
```

Comment:
Unused block argument - `controller`. You can omit the argument if you don't care about it.